Userland/Applications/PixelPaint/Tools/GradientTool.cpp

```diff
@@ -312,32 +364,59 @@ void GradientTool::draw_gradient(GUI::Painter& painter, bool with_guidelines, co
         Gfx::PainterStateSaver saver(painter);
         if (gradient_clip.has_value())
             painter.add_clip_rect(*gradient_clip);
-        painter.fill_rect_with_linear_gradient(gradient_rect, Array { Gfx::ColorStop { start_color, 0.5f - gradient_half_width_percentage_offset }, Gfx::ColorStop { end_color, 0.5f + gradient_half_width_percentage_offset } }, rotation_degrees);
+
+        if (m_mode == GradientMode::Linear)
+            painter.fill_rect_with_linear_gradient(gradient_rect, Array { Gfx::ColorStop { start_color, 0.5f - gradient_half_width_percentage_offset }, Gfx::ColorStop { end_color, 0.5f + gradient_half_width_percentage_offset } }, rotation_degrees);
+        else {
+            auto t_gradient_start = m_gradient_start.value().to_type<float>().scaled(scale, scale).translated(drawing_offset).to_type<int>();
+            auto t_gradient_left = m_gradient_horizontal_a.value().to_type<float>().scaled(scale, scale).translated(drawing_offset).to_type<int>();
```

Comment:
missed this name here; now i believe that `horizontal` is wrong and we should settle for `transversal` instead. as you rotate things thereʼs rarely left or right direction involved, also horizontal or vertical notions are bogus.

but you definitely have axes: _longitudinal_ and _transversal_ – at least this is what i see in the animation youʼve attached (i see no problems with start and end, though; `a` and `b` are suspicious but are way better than `left` and `right`). could you use that naming instead?

(p.s. note that `longitudinal` is still a proper name for linear gradient!)